sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/EncryptedBlobRange.java

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.cryptography;
+
+import com.azure.storage.blob.models.BlobRange;
+
+
+import static com.azure.storage.blob.cryptography.EncryptionConstants.ENCRYPTION_BLOCK_SIZE;
+
+/**
+ * This is a representation of a range of bytes on an encrypted blob, which may be expanded from the requested range to
+ * included extra data needed for encryption. Note that this type is not strictly thread-safe as the download method
+ * will update the count in case the user did not specify one. Passing null as an EncryptedBlobRange value will default
+ * to the entire range of the blob.
+ */
+final class EncryptedBlobRange {
+
+    /**
+     * The BlobRange passed by the customer and the range we must actually return.
+     */
+    private final BlobRange originalRange;
+
+    /**
+     * Amount the beginning of the range, 0-31, needs to be adjusted in order to align along an encryption block
+     * boundary and include the IV.
+     */
+    private final int offsetAdjustment;
+
+    /**
+     * How many bytes to download, including the adjustments for encryption block boundaries and the IV.
+     * Must be greater than or equal to 0 if specified.
+     */
+    private Long adjustedDownloadCount;
+
+    static EncryptedBlobRange getEncryptedBlobRangeFromHeader(String stringRange) {
+        // Null case
+        if (stringRange == null) {
+            return new EncryptedBlobRange(null);
+        }
+        // Non-null case
+        String trimmed = stringRange.substring(6); // Trim off the "bytes=" part
```

Comment:
Instead of using a magic number could we get the character position of the first `=`?